src/map/clif.c

```diff
@@ -20186,6 +20186,146 @@ void clif_skill_scale(struct block_list *bl, int src_id, int x, int y, uint16 sk
 #endif
 }
 
+bool clif_parse_attendance_db(void) {
+	struct config_t attendance_conf;
+	struct config_setting_t *attendance = NULL, *it = NULL;
+	const char *config_filename = "db/attendance_db.conf"; // FIXME hardcoded name
+	int i = 0;
+
+	if (!libconfig->load_file(&attendance_conf, config_filename))
+		return false;
+	attendance = libconfig->lookup(&attendance_conf, "attendance_db");
+
+	for (i = 0; i < MAX_ATTENDANCE_DAYS; i++) {
+		it = libconfig->setting_get_elem(attendance, i);
+		clif->attendancedb_libconfig_sub(it, i, config_filename);
+	}
+
+	libconfig->destroy(&attendance_conf);
+	ShowStatus("Done reading '"CL_WHITE"%d"CL_RESET"' entries in '"CL_WHITE"%s"CL_RESET"'.\n", i, config_filename);
+	return true;
+}
+
+bool clif_attendancedb_libconfig_sub(struct config_setting_t *it, int n, const char *source) {
+	struct config_setting_t *t = NULL;
```

Comment:
here need check it, source for null with nullpo
and check n for valid index with assertpo